### PR TITLE
Change all relative imports to absolute imports (except same-module imports)

### PR DESCRIPTION
**Pull request by astrofrog**
Absolute imports are generally more readable than e.g. 'from ...utils import x' and are recommended by PEP8. However, this keep imports from the same level (e.g. from .utils import ...) as relative imports. See https://github.com/astropy/astropy/issues/6452 for a discussion of this.

Fixes https://github.com/astropy/astropy/issues/6452

### CODE REVIEW

**Comment by bsipocz** [comment]
If this is a go ahead (for which I'm 👍), could the same PR also be used for the import reordering https://github.com/astropy/astropy/pull/7827 given that both would cause disruption everywhere...

**Comment by bsipocz** [comment]
(and I'm happy to sprint on the meeting to cleanup possible conflicts caused by this in other PRs during the meeting)

**Comment by eteq** [comment]
I was skeptical of this, because to me relative is *more* readable, makes it easier to move things, and catches some import problems... 

But out-of-band, @astrofrog and @taldcroft (and a few others) were persuasive and convinced me it's a good idea.  Because 1) Empirical evidence + PEP8 convinces me that I am the outlier in finding it more readable 2) it's not that common to move things and a search-and-replace does the job anyway, and 3) the py3.x absolute import business addresses most of those.

So in the end I'm 👍

**Comment by bsipocz** [comment]
@astrofrog - rebase than and merge away as soon as the release is out to minimize conflicts.

**Comment by mhvk** [comment]
Very much in favour! I did not look in great detail, but a suggestion in #6452 by @taldcroft was to have the in-module imports sit below the other astropy imports, i.e., effectively have four levels: python, other=numpy, other-astropy, local-to-module. Would the script be able to handle that? I see that, e.g., for `table`, this fourth level already exists, but for `units` it does not (only looked at `quantity.py`).

**Comment by astrofrog** [comment]
@mhvk - yes, @bsipocz also suggested reordering the imports. Since two of you now suggested that, I'll dig out my old script to do that...

**Comment by saimn** [comment]
I usually prefer local imports as it allows to easily see them, but here given the number of astropy imports at an higher level I agree that using absolute import is better.

Concerning sorting imports, if we are there I think that it should be enforced with something like isort, to be sure that it remains consistent in the future.  If there is an isort config then people can just configure their editor and use it.
I had a Go training last week, and Go has a very constraining set of tools and formatting rules (and a tool to automatically insert or remove imports!). This is annoying when you have other preferences, but this is also a huge time saver. No need to think about the formatting style when it is enforced.

**Comment by Cadair** [comment]
I wrote a isort config here: https://github.com/astropy/astropy/pull/7827/commits/88b423ec2fca60777d686b3eb9b03e9b7d908381

**Comment by astrofrog** [comment]
I just sorted the imports now but I'd like to open a separate PR, as we need to review things a bit more carefully (there are places where the imports are in a specific order for a reason).

So let's merge this first, then I'll open a follow-up.

**Comment by bsipocz** [comment]
@astrofrog - maybe that follow-up can also be done in chuckes per subpackage or a few subpackages per PR?

**Comment by pllim** [comment]
tl;dr -- Does developer doc need updating to reflect this new preference?

**Comment by astrofrog** [comment]
@pllim - see the last changed file (you need to scroll down a lot!)

File: docs/development/codeguide.rst [hunk 1]
```diff
@@ -143,11 +143,11 @@ Coding Style/Conventions
   maximum line length for different subpackages (typically either 80 or 100
   characters).  Please try to maintain the style when adding or modifying code.
 
-* One exception is to be made from the PEP8 style: new style relative imports
-  of the form ``from . import modname`` are allowed and required for Astropy,
-  as opposed to absolute (as PEP8 suggests) or the simpler ``import modname``
-  syntax. This is primarily due to improved relative import support since PEP8
-  was developed, and to simplify the process of moving modules.
+* One exception is to be made from the PEP8 style: while absolute imports
```
Comment (eteq) [inline docs/development/codeguide.rst]: This isn't really an exception, right? PEP8 says
> However, explicit relative imports are an acceptable alternative to absolute imports, especially when dealing with complex package layouts where using absolute imports would be unnecessarily verbose

I'll make a PR suggesting a re-word

**Comment by astrofrog** [comment]
@eteq - regarding the formatting of the imports, I'll fix that when reordering the imports in a separate PR.
